src/lib/y2partitioner/widgets/blk_devices_table.rb

```diff
@@ -263,66 +250,134 @@ def type_icon(device)
         thin:          N_("Thin LV")
       }
 
-      # Label for the device type (e.g., LVM, MD RAID)
+      # Label for the device and filesystem types (e.g., LVM, Ext4 RAID, Part of BtrFS /home, etc)
       #
-      # @param device [Y2Storage::BlkDevice]
+      # @param device [Y2Storage::Device]
       # @return [String]
       def type_label(device)
-        return default_type_label(device) unless device.is_a?(Y2Storage::BlkDevice)
+        return device.type.to_human_string if device.is?(:filesystem)
 
-        if device.is?(:partition)
-          partition_type_label(device)
+        fs = filesystem(device)
+        label = device_label(device)
+
+        if fs && fs.multidevice?
+          btrfs_multidevice_type_label(fs)
+        elsif fs
+          [fs.type.to_human_string, label].join(" ")
+        else
+          label
+        end
+      end
+
+      # Label for the device type (e.g., LVM, RAID, Partition, etc)
+      #
+      # @param device [Y2Storage::BlkDevice]
+      # @return [String]
+      def device_label(device)
+        if device.is?(:lvm_vg)
```

Comment:
Rubocop is complaining here because of `Perceived complexity`. And I agree with him :) but at the same time I think it isn't worth to split it. Any ideas?